mlflow/models/python_api.py

```diff
@@ -152,14 +168,55 @@ def predict(
         )
 
     """
+    # to avoid circular imports
+    from mlflow.pyfunc import (
+        _PREBUILD_ENV_ROOT_LOCATION,
+        _create_model_downloading_tmp_dir,
+        _gen_prebuilt_env_archive_name,
+    )
+
     if content_type not in [_CONTENT_TYPE_JSON, _CONTENT_TYPE_CSV]:
         raise MlflowException.invalid_parameter_value(
             f"Content type must be one of {_CONTENT_TYPE_JSON} or {_CONTENT_TYPE_CSV}."
         )
 
+    spark_session = None
+    if is_in_databricks_runtime():
+        try:
+            # if in Databricks, fetch the current active session
+            from databricks.sdk.runtime import spark
+
+            spark_session = spark
+        except Exception:
+            pass
+    is_dbconnect_mode = is_databricks_connect(spark_session)
+    if is_dbconnect_mode:
+        if env_manager != _EnvManager.VIRTUALENV:
+            raise MlflowException(
+                "Databricks Connect only supports virtualenv as the environment manager. "
+                f"Got {env_manager}."
+            )
+        local_model_path = _download_artifact_from_uri(
+            artifact_uri=model_uri,
+            output_path=_create_model_downloading_tmp_dir(should_use_nfs=False),
+        )
+        archive_name = _gen_prebuilt_env_archive_name(spark_session, local_model_path)
```

Comment:
Seemingly you don't need to call this function, what you need to do is just to put the env_root under /tmp/xxx directory